### PR TITLE
Default `RandomGraph` keeps num-edges fixed

### LINKED ISSUE #8: `RandomGraph` Default Behavior
## Bug / Design Issue: RandomGraph Example in Docs Fails Assertion

Our current documentation example for `RandomGraph` fails the expected assertion that the number of edges remains the same after transformation:

<details>
<summary>Example code from docs</summary>

```python
import torch
from torch_geometric.data import Data
from rings.perturbations import RandomGraph

# Create a simple graph
edge_index = torch.tensor([[0, 1, 1, 2], [1, 0, 2, 1]], dtype=torch.long)
data = Data(edge_index=edge_index, num_nodes=4)

# Example 1: Random graph with same number of edges
torch.manual_seed(42)  # For reproducibility
transform1 = RandomGraph()
t1_data = transform1(data.clone())
# Should have same number of edges but different structure
assert t1_data.edge_index.shape[1] == data.edge_index.shape[1]
assert not torch.equal(t1_data.edge_index, data.edge_index)
```

</details>

## Observed behavior

The first assertion fails because the transformed graph can have fewer edges. This is because our default Erdos-Rényi implementation drops self-loops, which reduces the number of edges.

Relevant code:
```python
# Remove self-loops
mask = row != col
row, col = row[mask], col[mask]
```

## Proposed solution
Refactor the self-loop removal logic, which is currently duplicated / ad hoc in places:
	•	Extract `_ensure_no_self_loops` into a standalone utility function.
	•	Reuse this utility consistently in perturbations.py, including `RandomGraph`, `ShuffleTransform`, and maybe future uses.

### PR DESCRIPTION
**Pull request by jeremy-wayland**
## Description
This PR adds a test case for the `RandomGraph` transform that reflects the example from our documentation. Previously we were dropping self loops when using an erdos reyni model– in order for better transparency we now ensure that we preserve the intended number of edges (i.e. make sure there are no self edges)

## Changes Made
- `esure_no_self_loops` is a standalon utility function used by both `RandomGraph` and `ShuffleTransform`
- Added `test_random_graph_transform_documentation_example()` to `test_perturbations.py`
- This test case exactly replicates the example from our documentation:
  - Creates a simple graph with the specified edge structure
  - Applies the `RandomGraph` transform with default parameters
  - Verifies that the transformed graph has the same number of edges
  - Ensures the edge structure is modified as expected
  - Confirms that node count and edge attributes are preserved correctly

## Why This Matters
Adding tests that mirror our documentation examples serves several important purposes:
1. **Documentation Accuracy**: Ensures our documentation examples are correct and actually work
2. **Regression Prevention**: Protects against future code changes breaking documented behavior
3. **Developer Experience**: Provides a reference implementation that developers can trust

## Testing Done
The new test passes successfully, confirming that the `RandomGraph` transform works as documented.

## Related Issues
Resolves #8: Add tests for documented examples

### CODE REVIEW

**Review by emsimons — APPROVED** [review]
Great! All tests passed.
